src/blocked.c

```diff
@@ -80,9 +77,42 @@ int getListPositionFromObjectOrReply(client *c, robj *arg, int *position);
  * on streams, as we have to remember the ID it blocked for. */
 typedef struct bkinfo {
     listNode *listnode;     /* List node for db->blocking_keys[key] list. */
-    streamID stream_id;     /* Stream ID if we blocked in a stream. */
```

Comment:
Should this just be a `ListNode listnode` so we don't have a double layer of indirection?